src/java/voldemort/client/protocol/admin/AdminClient.java

```diff
@@ -2016,67 +2016,79 @@ private String diffMessage(StoreDefinition newStoreDef, StoreDefinition remoteSt
 
         /**
          * Delete a store from all active nodes in the cluster
-         * <p>
          *
          * @param storeName name of the store to delete
+         * @throws VoldemortException of the first node which failed (note, there might be more)
+         * @see {@link #deleteStore(String, java.util.List)} for more visibility into specific failures.
          */
         public void deleteStore(String storeName) {
-            for(Node node: currentCluster.getNodes()) {
-                deleteStore(storeName, node.getId());
+            List<Integer> nodeIds = Lists.newArrayList(currentCluster.getNodeIds());
+            Map<Integer, VoldemortException> exceptionMap = deleteStore(storeName, nodeIds);
+            if (!exceptionMap.isEmpty()) {
+                throw exceptionMap.values().iterator().next();
```

Comment:
Sure, I could do that, but it's possible that there are different types of failures (i.e.: store doesn't exist on some nodes, while some other node is unreachable). What I did here is to maintain the previous behavior (throwing the "first" exception that happens), except that now there is a guarantee that it will have attempted the delete on every node, instead of stopping after the first failure.